modules/effects/arrow-effects-kotlinx-coroutines/src/main/kotlin/arrow/effects/DeferredK.kt

```diff
@@ -633,9 +773,17 @@ fun <A> DeferredKOf<A>.unsafeRunSync(): A =
  * ```
  */
 fun <A> DeferredKOf<A>.runAsync(cb: (Either<Throwable, A>) -> DeferredKOf<Unit>): DeferredK<Unit> =
-  DeferredK(scope(), Dispatchers.Unconfined, CoroutineStart.DEFAULT) {
-    fix().forceExceptionPropagation()
-    unsafeRunAsync(cb.andThen { it.unsafeRunAsync { } })
+  DeferredK(scope(), Dispatchers.Unconfined, CoroutineStart.LAZY) {
+    val result: A? = try {
+      await()
+    } catch (t: Throwable) {
+      cb(Left(t)).await()
+      null
+    }
+
+    result?.let { cb(Right(it)) }?.await()
+
+    Unit
```

Comment:
This shouldn't be necessary, right? Put elvish on the previous line, or `.let { }`

Reply:
Good catch